ddtrace/tracer/tracer.go

```diff
@@ -311,22 +314,42 @@ func (t *tracer) pushTrace(trace []*span) {
 	}
 }
 
-// StartSpan creates, starts, and returns a new Span with the given `operationName`.
+// StartSpan implements ddtrace.Tracer.
 func (t *tracer) StartSpan(operationName string, options ...ddtrace.StartSpanOption) ddtrace.Span {
+	span, _ := t.StartSpanFromContext(gocontext.Background(), operationName, options...)
+	return span
+}
+
+// StartSpanFromContext implements ddtrace.Tracer.
+func (t *tracer) StartSpanFromContext(ctx gocontext.Context, operationName string, options ...ddtrace.StartSpanOption) (ddtrace.Span, gocontext.Context) {
 	var opts ddtrace.StartSpanConfig
 	for _, fn := range options {
 		fn(&opts)
 	}
+	if ctx == nil {
+		ctx = gocontext.Background()
+	} else if s, ok := SpanFromContext(ctx); ok {
+		// span in ctx overwrite ChildOf() parent if any
+		opts.Parent = s.Context()
+	}
 	var startTime int64
 	if opts.StartTime.IsZero() {
 		startTime = now()
 	} else {
 		startTime = opts.StartTime.UnixNano()
 	}
 	var context *spanContext
+	pprofCtx := ctx
```

Comment:
If this is maintained completely separate from the user's context, and none of the user's context from it is used, why do we need it to be equal to it? Could it not be `context.Background()` just as good?

Reply:
The user context may contain [custom pprof labels](https://github.com/DataDog/dd-trace-go/blob/91efd989eb0b5f82d8bad78d5da16ea308d082f2/internal/traceprof/traceproftest/app.go#L210-L213) that we're trying to keep intact here.